brian2/codegen/cpp_prefs.py

```diff
@@ -239,7 +239,11 @@ def _determine_flag_compatibility(compiler, flagname):
     with tempfile.NamedTemporaryFile('w', suffix='.cpp') as f, std_silent():
         f.write('int main (int argc, char **argv) { return 0; }')
         try:
-            compiler.compile([f.name], extra_postargs=[flagname])
+            object_files = compiler.compile([f.name],
+                                            output_dir=os.path.dirname(f.name),
+                                            extra_postargs=[flagname])
+            for object_file in object_files:
```

Comment:
Just curious. Why's the unlink necessary? They should go out of scope by themselves?

Reply:
By "go out of scope" do you mean that they get automatically deleted when the `with` context manager ends? This is true for the `.cpp` file that we create, but the `.o` file (should be only one in this case) as the results of the compilation is not deleted automatically.

Reply:
Thanks. What about having a temporary directory instead? Then one would not have to unlink individual files?

Reply:
Ah, I see. Yes, that might be more elegant. Let me give it a quick try.